packages/mobile/src/screens/sign-on-screen/screens/ReviewHandleScreen.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useMemo } from 'react'
+
+import {
+  pickHandleSchema,
+  useAudiusQueryContext,
+  reviewHandlePageMessages as messages
+} from '@audius/common'
+import { setValueField } from 'audius-client/src/common/store/pages/signon/actions'
+import {
+  getCoverPhotoField,
+  getHandleField,
+  getLinkedSocialOnFirstPage,
+  getProfileImageField
+} from 'audius-client/src/common/store/pages/signon/selectors'
+import { Formik } from 'formik'
+import { useDispatch, useSelector } from 'react-redux'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
+
+import { Paper } from '@audius/harmony-native'
+import { TextField } from 'app/components/fields'
+import { useNavigation } from 'app/hooks/useNavigation'
+
+import { ReadOnlyAccountHeader } from '../components/AccountHeader'
+import { Heading, Page, PageFooter } from '../components/layout'
+import type { SignUpScreenParamList } from '../types'
+import { restrictedHandles } from '../utils/restrictedHandles'
+
+type ReviewHandleValues = {
+  handle: string
+}
+
+// Users land here if their handle doesn't work with our validation for some reason
+export const ReviewHandleScreen = () => {
+  const dispatch = useDispatch()
+  const { value: handle } = useSelector(getHandleField)
+  const { value: coverPhoto } = useSelector(getCoverPhotoField) ?? {}
+  const { value: profileImage } = useSelector(getProfileImageField) ?? {}
+  const hasImages = coverPhoto || profileImage
+  const initialValues = {
+    handle
+  }
+  const audiusQueryContext = useAudiusQueryContext()
+  const navigation = useNavigation<SignUpScreenParamList>()
+  const isLinkingSocialOnFirstPage = useSelector(getLinkedSocialOnFirstPage)
+
+  const validationSchema = useMemo(() => {
```

Comment:
small callout, we might want to useMemo the schema in create-login-details like this. ultimately prob doesnt matter much, but maybe for consistency?